spec/acceptance/organizer/iterate_spec.rb

```diff
@@ -1,44 +1,111 @@
 require 'spec_helper'
-require 'test_doubles'
+
+module IterateSpec
+  class AddsOneWithRollbackAction
+    extend LightService::Action
+    expects :number
+    promises :number
+
+    executed do |context|
+      context.number += 1
+    end
+
+    rolled_back do |context|
+      # puts "rolled_back AddsOne"
+      context.number -= 1
+    end
+  end
+
+  class AddsTwoWithRollbackAction
+    extend LightService::Action
+    expects :number
+    promises :number
+
+    executed do |context|
+      context.number += 2
+    end
+
+    rolled_back do |context|
+      # puts "rolled_back AddsTwo"
+      context.number -= 2
+    end
+  end
+
+  class FailsWithRollbackWhenReachesTwelve
+    extend LightService::Action
+    expects :number
+
+    executed do |context|
+      context.fail_with_rollback!("10 was reached, failing with rollback") \
+        if context.number >= 12
+    end
+  end
+
+  class TestIterate
+    extend LightService::Organizer
+
+    def self.call(context)
+      with(context)
+        .reduce([iterate(:counters,
+                         [AddsOneWithRollbackAction,
+                          AddsTwoWithRollbackAction,
+                          FailsWithRollbackWhenReachesTwelve])])
+    end
+
+    def self.call_single(context)
+      with(context)
+        .reduce([iterate(:counters,
+                         AddsOneWithRollbackAction)])
+    end
+  end
+end
```

Comment:
We already have `AddsOneWithRollbackAction` [here](https://github.com/adomokos/light-service/blob/e291a50f1d49a465fbc1ac82c3b490d426208406/spec/acceptance/rollback_spec.rb#L16-L30), I wonder if these could be merged and put to good use in `spec/test_doubles.rb`.